configapi/device_group_helpers.go

```diff
@@ -0,0 +1,291 @@
+			errorOccurred = true
+			continue
+		}
+		msg := &configmodels.ConfigMessage{
+			MsgMethod: configmodels.Post_op,
+			MsgType:   configmodels.Network_slice,
+			Slice:     &networkSlice,
+			SliceName: networkSlice.SliceName,
+		}
+		configChannel <- msg
+		logger.ConfigLog.Infof("network slice [%v] update sent to config channel", networkSlice.SliceName)
+	}
+	if errorOccurred {
+		return fmt.Errorf("one or more network slice updates failed (see logs)")
+	}
+	return nil
+}
+
+func deviceGroupPostHelper(requestDeviceGroup configmodels.DeviceGroups, msgOp int, groupName string) (int, error) {
+	logger.ConfigLog.Infof("received device group: %v", groupName)
+
+	ipdomain := &requestDeviceGroup.IpDomainExpanded
+	logger.ConfigLog.Infof("imsis.size: %v, Imsis: %v", len(requestDeviceGroup.Imsis), requestDeviceGroup.Imsis)
+	logger.ConfigLog.Infof("IP Domain Name: %v", requestDeviceGroup.IpDomainName)
+	logger.ConfigLog.Infof("IP Domain details: %v", ipdomain)
+	logger.ConfigLog.Infof("dnn name: %v", ipdomain.Dnn)
+	logger.ConfigLog.Infof("ue pool: %v", ipdomain.UeIpPool)
+	logger.ConfigLog.Infof("dns Primary: %v", ipdomain.DnsPrimary)
+	logger.ConfigLog.Infof("dns Secondary: %v", ipdomain.DnsSecondary)
+	logger.ConfigLog.Infof("ip mtu: %v", ipdomain.Mtu)
+	logger.ConfigLog.Infof("device Group Name: %v", groupName)
+
+	if ipdomain.UeDnnQos != nil {
+		ipdomain.UeDnnQos.DnnMbrDownlink = convertToBps(ipdomain.UeDnnQos.DnnMbrDownlink, ipdomain.UeDnnQos.BitrateUnit)
+		if ipdomain.UeDnnQos.DnnMbrDownlink < 0 {
+			ipdomain.UeDnnQos.DnnMbrDownlink = math.MaxInt64
+		}
+		logger.ConfigLog.Infof("MbrDownLink: %v", ipdomain.UeDnnQos.DnnMbrDownlink)
+		ipdomain.UeDnnQos.DnnMbrUplink = convertToBps(ipdomain.UeDnnQos.DnnMbrUplink, ipdomain.UeDnnQos.BitrateUnit)
+		if ipdomain.UeDnnQos.DnnMbrUplink < 0 {
+			ipdomain.UeDnnQos.DnnMbrUplink = math.MaxInt64
+		}
+		logger.ConfigLog.Infof("MbrUpLink: %v", ipdomain.UeDnnQos.DnnMbrUplink)
+	}
+
+	prevDevGroup := getDeviceGroupByName(groupName)
+	requestDeviceGroup.DeviceGroupName = groupName
+	if prevDevGroup == nil {
+		logger.ConfigLog.Infof("creating new device group %v", groupName)
+		statusCode, err := createDG(&requestDeviceGroup)
+		if err != nil {
+			return statusCode, err
+		}
+	} else {
+		statusCode, err := updateDG(&requestDeviceGroup, prevDevGroup)
+		if err != nil {
+			return statusCode, err
+		}
+	}
+
+	var msg configmodels.ConfigMessage
+	msg.MsgType = configmodels.Device_group
+	msg.MsgMethod = msgOp
+	msg.DevGroup = &requestDeviceGroup
+	msg.DevGroupName = groupName
+	configChannel <- &msg
+	logger.ConfigLog.Infof("successfully added Device Group [%v] to config channel", groupName)
+	return http.StatusOK, nil
+}
+
+func createDG(devGroup *configmodels.DeviceGroups) (int, error) {
+	if statusCode, err := handleDeviceGroupPost(devGroup, nil); err != nil {
+		logger.ConfigLog.Errorf("error creating device group %v: %v", devGroup, err)
```

Comment:
Should not this be `%+v` instead of `%v`?
```suggestion
		logger.ConfigLog.Errorf("error creating device group %+v: %+v", devGroup, err)
```